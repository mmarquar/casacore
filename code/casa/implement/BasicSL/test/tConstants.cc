//# tConstants.cc: This program tests the AIPS++ C (constants) class
//# Copyright (C) 1993,1994,1995,1999
//# Associated Universities, Inc. Washington DC, USA.
//#
//# This program is free software; you can redistribute it and/or modify it
//# under the terms of the GNU General Public License as published by the Free
//# Software Foundation; either version 2 of the License, or (at your option)
//# any later version.
//#
//# This program is distributed in the hope that it will be useful, but WITHOUT
//# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
//# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
//# more details.
//#
//# You should have received a copy of the GNU General Public License along
//# with this program; if not, write to the Free Software Foundation, Inc.,
//# 675 Massachusetts Ave, Cambridge, MA 02139, USA.
//#
//# Correspondence concerning AIPS++ should be addressed as follows:
//#        Internet email: aips2-request@nrao.edu.
//#        Postal address: AIPS++ Project Office
//#                        National Radio Astronomy Observatory
//#                        520 Edgemont Road
//#                        Charlottesville, VA 22903-2475 USA
//#
//# $Id$

//# Includes

#include <iostream.h>
#include <aips/Mathematics/Constants.h>

main()
{
   cout << ">>> List of machine constants."                 << endl
        << endl
        << "Machine constants from limits.h:"               << endl
        << "CHAR_BIT..............  " << CHAR_BIT           << endl
        << "CHAR_MIN..............  " << CHAR_MIN           << endl
        << "CHAR_MAX..............  " << CHAR_MAX           << endl
        << "SCHAR_MIN.............  " << SCHAR_MIN          << endl
        << "SCHAR_MAX.............  " << SCHAR_MAX          << endl
        << "UCHAR_MAX.............  " << UCHAR_MAX          << endl
        << "MB_LEN_MAX............  " << MB_LEN_MAX         << endl
        << "SHRT_MIN..............  " << SHRT_MIN           << endl
        << "SHRT_MAX..............  " << SHRT_MAX           << endl
        << "USHRT_MAX.............  " << USHRT_MAX          << endl
        << "INT_MIN...............  " << INT_MIN            << endl
        << "INT_MAX...............  " << INT_MAX            << endl
        << "UINT_MAX..............  " << UINT_MAX           << endl
        << "LONG_MIN..............  " << LONG_MIN           << endl
        << "LONG_MAX..............  " << LONG_MAX           << endl
        << "ULONG_MAX.............  " << ULONG_MAX          << endl;

   cout << endl
        << "Machine constants from float.h:"                << endl
        << "FLT_RADIX.............  " << FLT_RADIX          << endl
        << "FLT_ROUNDS............  " << FLT_ROUNDS         << endl
        << "FLT_MIN_EXP...........  " << FLT_MIN_EXP        << endl
        << "DBL_MIN_EXP...........  " << DBL_MIN_EXP        << endl
        << "LDBL_MIN_EXP..........  " << LDBL_MIN_EXP       << endl
        << "FLT_MAX_EXP...........  " << FLT_MAX_EXP        << endl
        << "DBL_MAX_EXP...........  " << DBL_MAX_EXP        << endl
        << "LDBL_MAX_EXP..........  " << LDBL_MAX_EXP       << endl
        << "FLT_MIN_10_EXP........  " << FLT_MIN_10_EXP     << endl
        << "DBL_MIN_10_EXP........  " << DBL_MIN_10_EXP     << endl
        << "LDBL_MIN_10_EXP.......  " << LDBL_MIN_10_EXP    << endl
        << "FLT_MAX_10_EXP........  " << FLT_MAX_10_EXP     << endl
        << "DBL_MAX_10_EXP........  " << DBL_MAX_10_EXP     << endl
        << "LDBL_MAX_10_EXP.......  " << LDBL_MAX_10_EXP    << endl
        << "FLT_MANT_DIG..........  " << FLT_MANT_DIG       << endl
        << "DBL_MANT_DIG..........  " << DBL_MANT_DIG       << endl
        << "LDBL_MANT_DIG.........  " << LDBL_MANT_DIG      << endl
        << "FLT_DIG...............  " << FLT_DIG            << endl
        << "DBL_DIG...............  " << DBL_DIG            << endl
        << "LDBL_DIG..............  " << LDBL_DIG           << endl
        << "FLT_EPSILON...........  " << FLT_EPSILON        << endl
        << "DBL_EPSILON...........  " << DBL_EPSILON        << endl
        << "LDBL_EPSILON..........  ";
                            if (LDBL_MAX_EXP == DBL_MAX_EXP) {
                                 cout << LDBL_EPSILON       << endl;
                            } else {
                                 cout << "(unprintable)"    << endl;
                            }
   cout << "FLT_MIN...............  " << FLT_MIN            << endl
        << "DBL_MIN...............  " << DBL_MIN            << endl
        << "LDBL_MIN..............  ";
                            if (LDBL_MAX_EXP == DBL_MAX_EXP) {
                                 cout << LDBL_MIN           << endl;
                            } else {
                                 cout << "(unprintable)"    << endl;
                            }
   cout << "FLT_MAX...............  " << FLT_MAX            << endl
        << "DBL_MAX...............  " << DBL_MAX            << endl
        << "LDBL_MAX..............  ";
                            if (LDBL_MAX_EXP == DBL_MAX_EXP) {
                                 cout << LDBL_MAX           << endl;
                            } else {
                                 cout << "(unprintable)"    << endl;
                            }
   cout << endl
        << "Machine constants from values.h:"               << endl
        << "HIBITS................  " << HIBITS             << endl
        << "HIBITL................  " << HIBITL             << endl
        << "MAXSHORT..............  " << MAXSHORT           << endl
        << "MAXLONG...............  " << MAXLONG            << endl
        << "MAXINT................  " << MAXINT             << endl
        << "MINFLOAT..............  " << MINFLOAT           << endl
        << "MINDOUBLE.............  " << MINDOUBLE          << endl
        << "MAXFLOAT..............  " << MAXFLOAT           << endl
        << "MAXDOUBLE.............  " << MAXDOUBLE          << endl;

   cout << endl
        << "<<< End of machine constants." << endl;

   cout << endl
        << "Irrationals:"                                   << endl
        << "C::sqrt2..............  " << C::sqrt2           << endl
        << "C::sqrt3..............  " << C::sqrt3           << endl
        << "C::_1_sqrt2...........  " << C::_1_sqrt2        << endl
        << "C::_1_sqrt3...........  " << C::_1_sqrt3        << endl;

   cout << endl
        << "Pi and functions thereof:"                      << endl
        << "C::pi.................  " << C::pi              << endl
        << "C::_2pi...............  " << C::_2pi            << endl
        << "C::pi_2...............  " << C::pi_2            << endl
        << "C::pi_4...............  " << C::pi_4            << endl
        << "C::_1_pi..............  " << C::_1_pi           << endl
        << "C::_2_pi..............  " << C::_2_pi           << endl
        << "C::_1_sqrtpi..........  " << C::_1_sqrtpi       << endl
        << "C::_2_sqrtpi..........  " << C::_2_sqrtpi       << endl;

   cout << endl
        << "e and functions thereof:"                       << endl
        << "C::e..................  " << C::e               << endl
        << "C::ln2................  " << C::ln2             << endl
        << "C::ln10...............  " << C::ln10            << endl
        << "C::log2e..............  " << C::log2e           << endl
        << "C::log10e.............  " << C::log10e          << endl;

   cout << endl
        << "gamma and functions thereof:"                   << endl
        << "C::gamma..............  " << C::gamma           << endl
        << "C::lngamma............  " << C::lngamma         << endl
        << "C::etogamma...........  " << C::etogamma        << endl;

   cout << endl
        << "Fundamental physical constants (SI units):"     << endl
        << "C::c..................  " << C::c               << endl
        << "C::Gravity............  " << C::Gravity         << endl
        << "C::Planck.............  " << C::Planck          << endl
        << "C::GasConst...........  " << C::GasConst        << endl
        << "C::Avogadro...........  " << C::Avogadro        << endl
        << "C::qe.................  " << C::qe              << endl
        << "C::mp.................  " << C::mp              << endl
        << "C::mp_me..............  " << C::mp_me           << endl;

   cout << endl
        << "Derived physical constants (SI units):"         << endl
        << "C::mu0................  " << C::mu0             << endl
        << "C::epsilon0...........  " << C::epsilon0        << endl
        << "C::Planck_2pi.........  " << C::Planck_2pi      << endl
        << "C::u..................  " << C::u               << endl
        << "C::Boltzmann..........  " << C::Boltzmann       << endl
        << "C::Faraday............  " << C::Faraday         << endl
        << "C::me.................  " << C::me              << endl
        << "C::re.................  " << C::re              << endl
        << "C::a0.................  " << C::a0              << endl;

   cout << endl
        << "Physical quantities (SI units):"                << endl
        << "C::R0.................  " << C::R0              << endl;

   cout << endl
        << "Metric prefixes:"                               << endl
        << "C::yotta..............  " << C::yotta           << endl
        << "C::zetta..............  " << C::zetta           << endl
        << "C::exa................  " << C::exa             << endl
        << "C::peta...............  " << C::peta            << endl
        << "C::tera...............  " << C::tera            << endl
        << "C::giga...............  " << C::giga            << endl
        << "C::mega...............  " << C::mega            << endl
        << "C::kilo...............  " << C::kilo            << endl
        << "C::hecto..............  " << C::hecto           << endl
        << "C::deka...............  " << C::deka            << endl
        << "C::deci...............  " << C::deci            << endl
        << "C::centi..............  " << C::centi           << endl
        << "C::milli..............  " << C::milli           << endl
        << "C::micro..............  " << C::micro           << endl
        << "C::nano...............  " << C::nano            << endl
        << "C::pico...............  " << C::pico            << endl
        << "C::femto..............  " << C::femto           << endl
        << "C::atto...............  " << C::atto            << endl
        << "C::zepto..............  " << C::zepto           << endl
        << "C::yocto..............  " << C::yocto           << endl;

   cout << endl
        << "Angular measure:"                               << endl
        << "C::radian.............  " << C::radian          << endl
        << "C::circle.............  " << C::circle          << endl
        << "C::circuit............  " << C::circuit         << endl
        << "C::cycle..............  " << C::cycle           << endl
        << "C::rev................  " << C::rev             << endl
        << "C::revolution.........  " << C::revolution      << endl
        << "C::rotation...........  " << C::rotation        << endl
        << "C::degree.............  " << C::degree          << endl
        << "C::arcmin.............  " << C::arcmin          << endl
        << "C::arcsec.............  " << C::arcsec          << endl 
        << "C::grad...............  " << C::grad            << endl;

   cout << endl
        << "Solid angular measure:"                         << endl
        << "C::steradian..........  " << C::steradian       << endl
        << "C::sphere.............  " << C::sphere          << endl
        << "C::square_degree......  " << C::square_degree   << endl
        << "C::square_arcmin......  " << C::square_arcmin   << endl
        << "C::square_arcsec......  " << C::square_arcsec   << endl;

   cout << endl
        << "Time interval:"                                 << endl
        << "C::second.............  " << C::second          << endl
        << "C::minute.............  " << C::minute          << endl
        << "C::hour...............  " << C::hour            << endl
        << "C::day................  " << C::day             << endl
        << "C::week...............  " << C::week            << endl
        << "C::fortnight..........  " << C::fortnight       << endl;

   cout << endl
        << "Frequency:"                                     << endl
        << "C::Hertz..............  " << C::Hertz           << endl;

   cout << endl
        << "Length:"                                        << endl
        << "C::metre..............  " << C::metre           << endl
        << "C::meter..............  " << C::meter           << endl
        << "C::Fermi..............  " << C::Fermi           << endl
        << "C::Angstrom...........  " << C::Angstrom        << endl
        << "C::inch...............  " << C::inch            << endl
        << "C::thou...............  " << C::thou            << endl
        << "C::hand...............  " << C::hand            << endl
        << "C::span...............  " << C::span            << endl
        << "C::foot...............  " << C::foot            << endl
        << "C::yard...............  " << C::yard            << endl
        << "C::fathom.............  " << C::fathom          << endl
        << "C::rod................  " << C::rod             << endl
        << "C::perch..............  " << C::perch           << endl
        << "C::rope...............  " << C::rope            << endl
        << "C::chain..............  " << C::chain           << endl
        << "C::furlong............  " << C::furlong         << endl
        << "C::mile...............  " << C::mile            << endl
        << "C::nautical_mile......  " << C::nautical_mile   << endl
        << "C::point..............  " << C::point           << endl
        << "C::pica...............  " << C::pica            << endl
        << "C::astronomical_unit..  " << C::astronomical_unit << endl
        << "C::light_second.......  " << C::light_second    << endl
        << "C::light_year.........  " << C::light_year      << endl
        << "C::parsec.............  " << C::parsec          << endl;

   cout << endl
        << "Area:"                                          << endl
        << "C::square_metre.......  " << C::square_metre    << endl
        << "C::square_meter.......  " << C::square_meter    << endl
        << "C::are................  " << C::are             << endl
        << "C::barn...............  " << C::barn            << endl
        << "C::square_inch........  " << C::square_inch     << endl
        << "C::square_foot........  " << C::square_foot     << endl
        << "C::square_yard........  " << C::square_yard     << endl
        << "C::square_mile........  " << C::square_mile     << endl
        << "C::square_perch.......  " << C::square_perch    << endl
        << "C::rood...............  " << C::rood            << endl
        << "C::acre...............  " << C::acre            << endl
        << "C::square.............  " << C::square          << endl;

   cout << endl
        << "Volume:"                                        << endl
        << "C::cubic_metre........  " << C::cubic_metre     << endl
        << "C::cubic_meter........  " << C::cubic_meter     << endl
        << "C::stere..............  " << C::stere           << endl
        << "C::litre..............  " << C::litre           << endl
        << "C::liter..............  " << C::liter           << endl
        << "C::cubic_inch.........  " << C::cubic_inch      << endl
        << "C::cubic_foot.........  " << C::cubic_foot      << endl
        << "C::cubic_yard.........  " << C::cubic_yard      << endl
        << "C::cubic_mile.........  " << C::cubic_mile      << endl
        << "C::gallon.............  " << C::gallon          << endl
        << "C::quart..............  " << C::quart           << endl
        << "C::pint...............  " << C::pint            << endl
        << "C::gill...............  " << C::gill            << endl
        << "C::fluid_ounce........  " << C::fluid_ounce     << endl
        << "C::drachm.............  " << C::drachm          << endl
        << "C::scruple............  " << C::scruple         << endl
        << "C::minim..............  " << C::minim           << endl
        << "C::USgallon...........  " << C::USgallon        << endl
        << "C::USquart............  " << C::USquart         << endl
        << "C::USpint.............  " << C::USpint          << endl
        << "C::USgill.............  " << C::USgill          << endl
        << "C::USfluid_ounce......  " << C::USfluid_ounce   << endl
        << "C::USdram.............  " << C::USdram          << endl
        << "C::USminim............  " << C::USminim         << endl;

   cout << endl
        << "Speed:"                                         << endl
        << "C::knot...............  " << C::knot            << endl;

   cout << endl
        << "Acceleration:"                                  << endl
        << "C::g..................  " << C::g               << endl;

   cout << endl
        << "Mass:"                                          << endl
        << "C::gram...............  " << C::gram            << endl
        << "C::tonne..............  " << C::tonne           << endl
        << "C::carat..............  " << C::carat           << endl
        << "C::pound..............  " << C::pound           << endl
        << "C::ounce..............  " << C::ounce           << endl
        << "C::stone..............  " << C::stone           << endl
        << "C::quarter............  " << C::quarter         << endl
        << "C::hundredweight......  " << C::hundredweight   << endl
        << "C::ton................  " << C::ton             << endl
        << "C::cental.............  " << C::cental          << endl
        << "C::shortquarter.......  " << C::shortquarter    << endl
        << "C::shortcwt...........  " << C::shortcwt        << endl
        << "C::shortton...........  " << C::shortton        << endl;

   cout << endl
        << "Force:"                                         << endl
        << "C::Newton.............  " << C::Newton          << endl
        << "C::dyne...............  " << C::dyne            << endl;

   cout << endl
        << "Pressure:"                                      << endl
        << "C::Pascal.............  " << C::Pascal          << endl
        << "C::atmosphere.........  " << C::atmosphere      << endl
        << "C::bar................  " << C::bar             << endl
        << "C::torr...............  " << C::torr            << endl
        << "C::mmHg...............  " << C::mmHg            << endl;

   cout << endl
        << "Energy:"                                        << endl
        << "C::Joule..............  " << C::Joule           << endl
        << "C::kWh................  " << C::kWh             << endl
        << "C::erg................  " << C::erg             << endl
        << "C::calorie............  " << C::calorie         << endl
        << "C::calorie_IT.........  " << C::calorie_IT      << endl
        << "C::Btu................  " << C::Btu             << endl
        << "C::eV.................  " << C::eV              << endl;

   cout << endl
        << "Temperature difference:"                        << endl
        << "C::Kelvin.............  " << C::Kelvin          << endl
        << "C::Celsius............  " << C::Celsius         << endl
        << "C::Centigrade.........  " << C::Centigrade      << endl
        << "C::Fahrenheit.........  " << C::Fahrenheit      << endl
        << "C::Rankine............  " << C::Rankine         << endl;

   cout << endl
        << "Temperature at 0 on each temperature scale:"    << endl
        << "C::Kelvin_0...........  " << C::Kelvin_0        << endl
        << "C::Celsius_0..........  " << C::Celsius_0       << endl
        << "C::Centigrade_0.......  " << C::Centigrade_0    << endl
        << "C::Fahrenheit_0.......  " << C::Fahrenheit_0    << endl
        << "C::Rankine_0..........  " << C::Rankine_0       << endl;

   cout << endl
        << "Power:"                                         << endl
        << "C::Watt...............  " << C::Watt            << endl
        << "C::horsepower.........  " << C::horsepower      << endl;

   cout << endl
        << "Flux density:"                                  << endl
        << "C::Jansky.............  " << C::Jansky          << endl
        << "C::fu.................  " << C::fu              << endl;

   cout << endl
        << "Electric charge:"                               << endl
        << "C::Coulomb............  " << C::Coulomb         << endl
        << "C::abCoulomb..........  " << C::abCoulomb       << endl
        << "C::statCoulomb........  " << C::statCoulomb     << endl;

   cout << endl
        << "Electric current:"                              << endl
        << "C::Ampere.............  " << C::Ampere          << endl
        << "C::abAmpere...........  " << C::abAmpere        << endl
        << "C::statAmpere.........  " << C::statAmpere      << endl;

   cout << endl
        << "Electric potential:"                            << endl
        << "C::Volt...............  " << C::Volt            << endl
        << "C::abVolt.............  " << C::abVolt          << endl
        << "C::statVolt...........  " << C::statVolt        << endl;

   cout << endl
        << "Electric resistance:"                           << endl
        << "C::Ohm................  " << C::Ohm             << endl
        << "C::abOhm..............  " << C::abOhm           << endl
        << "C::statOhm............  " << C::statOhm         << endl;

   cout << endl
        << "Electric conductance:"                          << endl
        << "C::Siemens............  " << C::Siemens         << endl
        << "C::mho................  " << C::mho             << endl;

   cout << endl
        << "Electric capacitance:"                          << endl
        << "C::Farad..............  " << C::Farad           << endl
        << "C::abFarad............  " << C::abFarad         << endl
        << "C::statFarad..........  " << C::statFarad       << endl;

   cout << endl
        << "Electric inductance:"                           << endl
        << "C::Henry..............  " << C::Henry           << endl
        << "C::abHenry............  " << C::abHenry         << endl
        << "C::statHenry..........  " << C::statHenry       << endl;

   cout << endl
        << "Magnetic induction:"                            << endl
        << "C::Tesla..............  " << C::Tesla           << endl
        << "C::Gauss..............  " << C::Gauss           << endl;

   cout << endl
        << "Magnetic flux:"                                 << endl
        << "C::Weber..............  " << C::Weber           << endl
        << "C::Maxwell............  " << C::Maxwell         << endl
        << "C::line...............  " << C::line            << endl;

   cout << endl
        << "Magnetomotance:"                                << endl
        << "C::Ampere_turn........  " << C::Ampere_turn     << endl
        << "C::abAmpere_turn......  " << C::abAmpere_turn   << endl
        << "C::Gilbert............  " << C::Gilbert         << endl
        << "C::praGilbert.........  " << C::praGilbert      << endl;

   cout << endl
        << "Magnetic field intensity:"                      << endl
        << "C::Oersted............  " << C::Oersted         << endl
        << "C::praOersted.........  " << C::praOersted      << endl;

   cout << endl
        << "Radioactivity:"                                 << endl
        << "C::Bequerel...........  " << C::Bequerel        << endl;

   cout << endl
        << "Luminous intensity:"                            << endl
        << "C::candela............  " << C::candela         << endl;

   cout << endl
        << "Amount of substance:"                           << endl
        << "C::mole...............  " << C::mole            << endl
        << "C::molecule...........  " << C::molecule        << endl;

   return 0;
}
