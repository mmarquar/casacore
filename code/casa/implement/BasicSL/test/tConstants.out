>>> List of machine constants.

Machine constants from limits.h:
CHAR_BIT..............  8
CHAR_MIN..............  -128
CHAR_MAX..............  127
SCHAR_MIN.............  -128
SCHAR_MAX.............  127
UCHAR_MAX.............  255
MB_LEN_MAX............  5
SHRT_MIN..............  -32768
SHRT_MAX..............  32767
USHRT_MAX.............  65535
INT_MIN...............  -2147483648
INT_MAX...............  2147483647
UINT_MAX..............  4294967295
LONG_MIN..............  -2147483648
LONG_MAX..............  2147483647
ULONG_MAX.............  4294967295

Machine constants from float.h:
FLT_RADIX.............  2
FLT_MIN_EXP...........  -125
DBL_MIN_EXP...........  -1021
LDBL_MIN_EXP..........  -16381
FLT_MAX_EXP...........  128
DBL_MAX_EXP...........  1024
LDBL_MAX_EXP..........  16384
FLT_MIN_10_EXP........  -37
DBL_MIN_10_EXP........  -307
LDBL_MIN_10_EXP.......  -4931
FLT_MAX_10_EXP........  38
DBL_MAX_10_EXP........  308
LDBL_MAX_10_EXP.......  4932
FLT_MANT_DIG..........  24
DBL_MANT_DIG..........  53
LDBL_MANT_DIG.........  113
FLT_DIG...............  6
DBL_DIG...............  15
LDBL_DIG..............  33
FLT_EPSILON...........  1.19209e-07
DBL_EPSILON...........  2.22045e-16
LDBL_EPSILON..........  (unprintable)
FLT_MIN...............  1.17549e-38
DBL_MIN...............  2.22507e-308
LDBL_MIN..............  (unprintable)
FLT_MAX...............  3.40282e+38
DBL_MAX...............  1.79769e+308
LDBL_MAX..............  (unprintable)

Machine constants from values.h:
BITS(Char)............  8
BITS(uChar)...........  8
BITS(Short)...........  16
BITS(uShort)..........  16
BITS(Int).............  32
BITS(uInt)............  32
BITS(Float)...........  32
BITS(Double)..........  64
BITS(lDouble).........  128
HIBITS................  -32768
HIBITL................  2147483648
HIBITI................  2147483648
MAXSHORT..............  32767
MAXLONG...............  2147483647
MAXINT................  2147483647
MINFLOAT..............  1.4013e-45
LN_MINFLOAT...........  -103.279
MINDOUBLE.............  4.94066e-324
LN_MINDOUBLE..........  -744.44
MAXFLOAT..............  3.40282e+38
LN_MAXFLOAT...........  88.7228
MAXDOUBLE.............  1.79769e+308
LN_MAXDOUBLE..........  709.783
FSIGNIF...............  24
DSIGNIF...............  53

<<< End of machine constants.

Irrationals:
C::sqrt2..............  1.41421
C::sqrt3..............  1.73205
C::_1_sqrt2...........  0.707107
C::_1_sqrt3...........  0.57735

Pi and functions thereof:
C::pi.................  3.14159
C::_2pi...............  6.28319
C::pi_2...............  1.5708
C::pi_4...............  0.785398
C::_1_pi..............  0.31831
C::_2_pi..............  0.63662
C::_1_sqrtpi..........  0.56419
C::_2_sqrtpi..........  1.12838

e and functions thereof:
C::e..................  2.71828
C::ln2................  0.693147
C::ln10...............  2.30259
C::log2e..............  1.4427
C::log10e.............  0.434294

gamma and functions thereof:
C::gamma..............  0.577216
C::lngamma............  -0.549539
C::etogamma...........  1.78107

Fundamental physical constants (SI units):
C::c..................  2.99792e+08
C::Gravity............  6.67259e-11
C::Planck.............  6.62608e-34
C::GasConst...........  8.31451
C::Avogadro...........  6.02214e+23
C::qe.................  1.60218e-19
C::mp.................  1.67262e-27
C::mp_me..............  1836.15

Derived physical constants (SI units):
C::mu0................  1.25664e-06
C::epsilon0...........  8.85419e-12
C::Planck_2pi.........  1.05457e-34
C::u..................  1.66054e-27
C::Boltzmann..........  1.38066e-23
C::Faraday............  96485.3
C::me.................  9.10939e-31
C::re.................  2.8179e-15
C::a0.................  5.2918e-11

Physical quantities (SI units):
C::R0.................  6.9599e+08

Metric prefixes:
C::yotta..............  1e+24
C::zetta..............  1e+21
C::exa................  1e+18
C::peta...............  1e+15
C::tera...............  1e+12
C::giga...............  1e+09
C::mega...............  1e+06
C::kilo...............  1000
C::hecto..............  100
C::deka...............  10
C::deci...............  0.1
C::centi..............  0.01
C::milli..............  0.001
C::micro..............  1e-06
C::nano...............  1e-09
C::pico...............  1e-12
C::femto..............  1e-15
C::atto...............  1e-18
C::zepto..............  1e-21
C::yocto..............  1e-24

Angular measure:
C::radian.............  1
C::circle.............  6.28319
C::circuit............  6.28319
C::cycle..............  6.28319
C::rev................  6.28319
C::revolution.........  6.28319
C::rotation...........  6.28319
C::degree.............  0.0174533
C::arcmin.............  0.000290888
C::arcsec.............  4.84814e-06
C::grad...............  0.015708

Solid angular measure:
C::steradian..........  1
C::sphere.............  12.5664
C::square_degree......  0.000304617
C::square_arcmin......  8.46159e-08
C::square_arcsec......  2.35044e-11

Time interval:
C::second.............  1
C::minute.............  60
C::hour...............  3600
C::day................  86400
C::week...............  604800
C::fortnight..........  1.2096e+06

Frequency:
C::Hertz..............  1

Length:
C::metre..............  1
C::meter..............  1
C::Fermi..............  1e-15
C::Angstrom...........  1e-10
C::inch...............  0.0254
C::thou...............  2.54e-05
C::hand...............  0.1016
C::span...............  0.2286
C::foot...............  0.3048
C::yard...............  0.9144
C::fathom.............  1.8288
C::rod................  5.0292
C::perch..............  5.0292
C::rope...............  6.096
C::chain..............  20.1168
C::furlong............  201.168
C::mile...............  1609.34
C::nautical_mile......  1853.18
C::point..............  0.000352778
C::pica...............  0.00423333
C::astronomical_unit..  1.49598e+11
C::light_second.......  2.99792e+08
C::light_year.........  9.46073e+15
C::parsec.............  3.08568e+16

Area:
C::square_metre.......  1
C::square_meter.......  1
C::are................  100
C::barn...............  1e-28
C::square_inch........  0.00064516
C::square_foot........  0.092903
C::square_yard........  0.836127
C::square_mile........  2.58999e+06
C::square_perch.......  25.2929
C::rood...............  1011.71
C::acre...............  4046.86
C::square.............  9.2903

Volume:
C::cubic_metre........  1
C::cubic_meter........  1
C::stere..............  1
C::litre..............  0.001
C::liter..............  0.001
C::cubic_inch.........  1.63871e-05
C::cubic_foot.........  0.0283168
C::cubic_yard.........  0.764555
C::cubic_mile.........  4.16818e+09
C::gallon.............  0.00454609
C::quart..............  0.00113652
C::pint...............  0.000568261
C::gill...............  0.000142065
C::fluid_ounce........  2.8413e-05
C::drachm.............  3.55163e-06
C::scruple............  1.18388e-06
C::minim..............  5.91939e-08
C::USgallon...........  0.00378541
C::USquart............  0.000946353
C::USpint.............  0.000473176
C::USgill.............  0.000118294
C::USfluid_ounce......  2.95735e-05
C::USdram.............  3.69669e-06
C::USminim............  6.16115e-08

Speed:
C::knot...............  0.514773

Acceleration:
C::g..................  9.80665

Mass:
C::gram...............  0.001
C::tonne..............  1000
C::carat..............  0.0002
C::pound..............  0.453592
C::ounce..............  0.0283495
C::stone..............  6.35029
C::quarter............  12.7006
C::hundredweight......  50.8023
C::ton................  1016.05
C::cental.............  45.3592
C::shortquarter.......  11.3398
C::shortcwt...........  45.3592
C::shortton...........  907.185

Force:
C::Newton.............  1
C::dyne...............  1e-05

Pressure:
C::Pascal.............  1
C::atmosphere.........  101325
C::bar................  100000
C::torr...............  133.322
C::mmHg...............  133.322

Energy:
C::Joule..............  1
C::kWh................  3.6e+06
C::erg................  1e-07
C::calorie............  4.184
C::calorie_IT.........  4.1868
C::Btu................  1054.35
C::eV.................  1.60218e-19

Temperature difference:
C::Kelvin.............  1.38066e-23
C::Celsius............  1.38066e-23
C::Centigrade.........  1.38066e-23
C::Fahrenheit.........  7.67032e-24
C::Rankine............  7.67032e-24

Temperature at 0 on each temperature scale:
C::Kelvin_0...........  0
C::Celsius_0..........  3.77127e-21
C::Centigrade_0.......  3.77127e-21
C::Fahrenheit_0.......  3.52582e-21
C::Rankine_0..........  0

Power:
C::Watt...............  1
C::horsepower.........  745.7

Flux density:
C::Jansky.............  1e-26
C::fu.................  1e-26

Electric charge:
C::Coulomb............  1
C::abCoulomb..........  10
C::statCoulomb........  3.33564e-10

Electric current:
C::Ampere.............  1
C::abAmpere...........  10
C::statAmpere.........  3.33564e-10

Electric potential:
C::Volt...............  1
C::abVolt.............  1e-08
C::statVolt...........  299.792

Electric resistance:
C::Ohm................  1
C::abOhm..............  1e-09
C::statOhm............  8.99377e+11

Electric conductance:
C::Siemens............  1
C::mho................  1

Electric capacitance:
C::Farad..............  1
C::abFarad............  1e+09
C::statFarad..........  1.11188e-12

Electric inductance:
C::Henry..............  1
C::abHenry............  1e-09
C::statHenry..........  8.99377e+11

Magnetic induction:
C::Tesla..............  1
C::Gauss..............  0.0001

Magnetic flux:
C::Weber..............  1
C::Maxwell............  1e-08
C::line...............  1e-08

Magnetomotance:
C::Ampere_turn........  1
C::abAmpere_turn......  10
C::Gilbert............  0.795775
C::praGilbert.........  12.5664

Magnetic field intensity:
C::Oersted............  79.5775
C::praOersted.........  12.5664

Radioactivity:
C::Bequerel...........  1

Luminous intensity:
C::candela............  1

Amount of substance:
C::mole...............  1
C::molecule...........  1.66054e-24
